lxc/remote.go

```diff
@@ -326,6 +327,38 @@ func (c *cmdRemoteAdd) Run(cmd *cobra.Command, args []string) error {
 		}
 
 		conf.Remotes[server] = config.Remote{Addr: addr, Public: true, Protocol: c.flagProtocol}
+
+		// Test address / port connectivity
+		dialAddr := remoteURL.Hostname()
+		if remoteURL.Port() == "" {
+			dialAddr = dialAddr + ":443"
+		} else {
+			dialAddr = dialAddr + ":" + remoteURL.Port()
+		}
+
+		timeout := 5 * time.Second
+		_, err = net.DialTimeout("tcp", dialAddr, timeout)
```

Comment:
why do we need the DialTimeout here, rather than only doing the d.GetImages() call, which would also test the connectivity at the same time?